File: Phyton/Ejercicio2_2.py
# Python program to check if two strings are 
# Permutations of each other
NO_OF_CHARS = 256

# Function to check whether two strings are
# Permutation of each other
def arePermutation(str1, str2):
	count1 = [0] * NO_OF_CHARS
	count2 = [0] * NO_OF_CHARS
	for i in str1:
		count1[ord(i)]+=1

	for i in str2:
		count2[ord(i)]+=1

	if len(str1) != len(str2):
		return 0

	for i in range(NO_OF_CHARS):
		if count1[i] != count2[i]:
			return 0

	return 1

print("INGRESA DOS CADENAS ")
str1 = input()
str2 = input() 

if arePermutation(str1, str2):
	print("Yes")
else:
	print("No")

File: Java/Ejercicio2_2.java
// JAVA program to check if two strings
// are Permutations of each other
import java.util.*;

class GFG{
	
	static int NO_OF_CHARS = 256;
	static boolean arePermutation(char str1[], char str2[]){
		int count1[] = new int [NO_OF_CHARS];
		Arrays.fill(count1, 0);
		int count2[] = new int [NO_OF_CHARS];
		Arrays.fill(count2, 0);
		int i;

		for (i = 0; i <str1.length && i < str2.length ; i++) {
			count1[str1[i]]++;
			count2[str2[i]]++;
		}
		if (str1.length != str2.length)
			return false;
		if (count1[i] != count2[i])
				return false;

		return true;
	} 

	public static void main(String args[]){
		char str1[] = ("geeksforgeeks").toCharArray();
		char str2[] = ("forgeeksgeeks").toCharArray();
		
		if ( arePermutation(str1, str2) )
			System.out.println("Yes");
		else
			System.out.println("No");
	}
}
